code/modules/interview/interview.dm

```diff
@@ -0,0 +1,170 @@
+	var/read_only = FALSE
+	/// Integer that contains the current position in the interview queue, used for rendering
+	var/pos_in_queue
+	/// Contains the state of the form, used for rendering and sanity checking
+	var/status = INTERVIEW_PENDING
+
+/datum/interview/New(client/interviewee)
+	if(!interviewee)
+		qdel(src)
+		return
+	id = ++atomic_id
+	owner = interviewee
+	owner_ckey = owner.ckey
+	questions = CONFIG_GET(str_list/interview_questions)
+	responses.len = questions.len
+	welcome_message = CONFIG_GET(string/interview_welcome_msg)
+
+/**
+  * Approves the interview, forces reconnect of owner if relevant.
+  *
+  * Approves the interview, and if relevant will force the owner to reconnect so that they have the proper
+  * verbs returned to them.
+  * Arguments:
+  * * approved_by - The user who approved the interview, used for logging
+  */
+/datum/interview/proc/approve(client/approved_by)
+	status = INTERVIEW_APPROVED
+	read_only = TRUE
+	GLOB.interviews.approved_ckeys |= owner_ckey
+	GLOB.interviews.close_interview(src)
+	log_admin_private("[key_name(approved_by)] has approved interview #[id] for [owner_ckey][!owner ? "(DC)": ""].")
+	message_admins("<span class='adminnotice'>[key_name(approved_by)] has approved interview #[id] for [owner_ckey][!owner ? "(DC)": ""].</span>")
+	if (owner)
+		SEND_SOUND(owner, sound('sound/effects/adminhelp.ogg'))
+		to_chat(owner, "<font color='red' size='4'><b>-- Interview Update --</b></font>" \
+			+ "\n<span class='adminsay'>Your interview was approved, you will now be reconnected in 5 seconds.</span>")
+		addtimer(CALLBACK(src, .proc/reconnect_owner), 50)
+
+/**
+  * Denies the interview and adds the owner to the cooldown for new interviews.
+  *
+  * Arguments:
+  * * denied_by - The user who denied the interview, used for logging
+  */
+/datum/interview/proc/deny(client/denied_by)
+	status = INTERVIEW_DENIED
+	read_only = TRUE
+	GLOB.interviews.close_interview(src)
+	GLOB.interviews.denied_ckeys |= owner_ckey
+	var/can_retry = CONFIG_GET(flag/panic_bunker_interview_retries)
+	log_admin_private("[key_name(denied_by)] has denied interview #[id] for [owner_ckey][!owner ? "(DC)": ""]. Retries [can_retry ? "enabled" : "disabled"].")
+	message_admins("<span class='adminnotice'>[key_name(denied_by)] has denied interview #[id] for [owner_ckey][!owner ? "(DC)": ""]. Retries [can_retry ? "enabled" : "disabled"].</span>")
+	if(can_retry)
+		addtimer(CALLBACK(GLOB.interviews, /datum/interview_manager.proc/release_from_cooldown, owner_ckey), 180 SECONDS)
+		if (owner)
+			SEND_SOUND(owner, sound('sound/effects/adminhelp.ogg'))
+			to_chat(owner, "<font color='red' size='4'><b>-- Interview Update --</b></font>" \
+				+ "\n<span class='adminsay'>Unfortunately your interview was denied. Please try submitting another questionnaire." \
+				+ " You may do this in three minutes.</span>")
+	else
+		addtimer(CALLBACK(GLOB.interviews, /datum/interview_manager.proc/give_the_boot, owner_ckey), 60 SECONDS)
```

Comment:
imo this should be shorter, like 30 seconds, no need to keep him on the server that long